### PR TITLE
Run copy_from_upstream and test

### LINKED ISSUE #1586: copy_from_upstream run mistake (also in 0.9.0 release)
When looking into https://github.com/open-quantum-safe/liboqs/issues/1584 I noticed that "main" (and 0.9.0) are not idempotent to a run of `copy_from_upstream` on my machine: See the differences in https://github.com/open-quantum-safe/liboqs/blob/mb-fixcopyfromupstream: Compare for example the documentation of algorithm feature sets for Sphincs.

We have [a test protecting against such failures](https://github.com/open-quantum-safe/liboqs/blob/bac716cc91c53529e3fad88e92ff9dfd19c04577/.github/workflows/linux.yml#L21-L36), but this "silently" [failed](https://github.com/open-quantum-safe/liboqs/actions/runs/6564959643/job/17832426451) i.e., incorrectly returned OK after reporting a failure. This ought to be investigated I'd gather.

Also the [Release process documentation](https://github.com/open-quantum-safe/liboqs/wiki/Release-process) was lacking in this regard (now fixed).

Could someone else please take a look (maybe first just confirm this finding as I'm not sure my machine is not a bit misconfigured), maybe @praveksharma or @SWilson4 ? If confirmed, we might want to add wording to the release notes of 0.9.0 that the algorithm documentation isn't "quite right": At first blush, it seems upstream Sphincs claims support for all OSs while `liboqs` only documents support for Linux&Darwin for optimized Sphincs... Or is this related to the limitations of our shared code in Windows? But why then doesn't `copy_from_upstream` reconcile this?

### PR DESCRIPTION
**Pull request by baentsch**
Fixes #1586

* [no] Does this PR change the input/output behaviour of a cryptographic algorithm (i.e., does it change known answer test values)?  (If so, a version bump will be required from *x.y.z* to *x.(y+1).0*.)
* [no] Does this PR change the list of algorithms available -- either adding, removing, or renaming? Does this PR otherwise change an API? (If so, PRs in fully supported downstream projects dependent on these, i.e., [oqs-provider](https://github.com/open-quantum-safe/oqs-provider) and [OQS-OpenSSH](https://github.com/open-quantum-safe/openssh) will also need to be ready for review and merge by the time this is merged.)

### CODE REVIEW

**Comment by baentsch** [comment]
@SWilson4 @praveksharma @dstebila Please carefully check this: I'm less than certain that the docs have all been generated correctly. The CI test improvement however definitely is needed.

File: docs/algorithms/sig/sphincs.yml
Comment (SWilson4) [inline docs/algorithms/sig/sphincs.yml:59]: I think we do want to document the restriction to Linux and Darwin: see [here](https://github.com/open-quantum-safe/liboqs/pull/1478).

**Review by SWilson4 — COMMENTED** [review]
I'd like to hear @praveksharma's thoughts on the constant-time documentation changes for McEliece and Falcon; if I recall correctly he recently worked on some related issues.

**Comment by baentsch** [comment]
Oops -- here's a PR I forgot to mention when we just discussed: @praveksharma @SWilson4 : Do you want to review or shall this be superseded by #1585? Would be totally fine with me. Main thing is #1586 gets fixed (incl. the CI shortcoming addressed in this PR).

**Comment by SWilson4** [comment]
> Oops -- here's a PR I forgot to mention when we just discussed: @praveksharma @SWilson4 : Do you want to review or shall this be superseded by #1585? Would be totally fine with me. Main thing is #1586 gets fixed (incl. the CI shortcoming addressed in this PR).

I'd rather not try to cram the CI fix into (the already massive) #1585, so on that front I'm OK with proceeding with this PR, once we're satisfied that the documentation is indeed generated correctly. _However_, once #1585 lands we will be out of sync with PQClean, so the CI fix should break the build. (Or, if this PR lands first, #1585 will be blocked from merging.)

How ought we to deal with this? Ideally. I think we'd resolve #1584, but that might get fairly involved. Still, it will need to be done if idempotence under `copy_from_upstream.py` is a desired property. We could also do a hacky workaround like disabling the CI check for now, or adding some exceptions to it, but that seems like a recipe for future pain.

**Comment by praveksharma** [comment]
> I'd like to hear @praveksharma's thoughts on the constant-time documentation changes for McEliece and Falcon; if I recall correctly he recently worked on some related issues.

@SWilson4, copying a comment from another PR over here: `classic_mceliece.md` wasn't updated along with `classic_mceliece.yml` in #1552 (presumably I forgot to run `update_upstream_alg_docs.py` at the time) so these changes are desirable. Ditto with Falcon.

**Comment by baentsch** [comment]
>  if idempotence under copy_from_upstream.py is a desired property.

I'd argue idempotence under `copy_from_upstream` is essential for this project: `liboqs` is pretty much completely gutted if it cannot reliably import the algorithms. As we're not under any time pressure, I'd argue to do it "right": Fix #1584 first, maybe (just as a test) use that to re-generate #1586 (it seems we now have clarity on the McEliece docs, thanks @praveksharma ), then re-base #1585 (giving the improved copy_from_upstream a final "run for its money" -- and have it focus on HQC and not the other shenenigans).

File: docs/algorithms/sig/sphincs.yml
Comment (baentsch) [reply docs/algorithms/sig/sphincs.yml]: With the latest commit this is now as desired (spec'd by the patch). "Reason": I ran `copy_from_upstream` with parameter `-k`: This way, the patches do not get thrown away as per [this code](https://github.com/open-quantum-safe/liboqs/blob/bac716cc91c53529e3fad88e92ff9dfd19c04577/scripts/copy_from_upstream/copy_from_upstream.py#L614-L615). Again, no clue why this `rmtree` statement exists... Unless I hear objections from @bhess I'll probably simply remove this line when resolving #1584 to avoid future repeats of this issue (documentation patches simply getting nixed).

So this PR is now ready for review @SWilson4 @praveksharma .

**Review by SWilson4 — APPROVED** [review]

File: docs/algorithms/sig/sphincs.yml
Comment (bhess) [reply docs/algorithms/sig/sphincs.yml]: @baentsch the `rmtree` was originally at the end of the `def copy_from_upstream` function with the intend to delete the "repos" directory if one doesn't want to keep it. 
The `update_upstream_alg_docs.do_it(os.environ['LIBOQS_DIR'])` seems to have been added later mistakenly after removing the repos.
I'd just fix this by moving `rmtree` again to the end of the function, otherwise the -k switch doesn't have any effect.

**Review by bhess — CHANGES_REQUESTED** [review]
see single comment above about `rmtree`.

**Comment by baentsch** [comment]
@dstebila Your "code owner" flag apparently overrules 3 people agreeing that this is OK, so please review.

**Review by dstebila — APPROVED** [review]

**Comment by dstebila** [comment]
> @dstebila Your "code owner" flag apparently overrules 3 people agreeing that this is OK, so please review.

I've approved it.  I've also changed the branch protection rules on `main` to require 2 committers to approve, rather than requiring 1 committer and a code owner, since we aren't consistently making use of the code owner functionality.

**Comment by baentsch** [comment]
@SWilson4 unless I hear howls of protest I'll merge this then and re-base "sw-hqc-pull"/#1585 as I'm already using that branch to test out my ideas for resolving #1584...

**Pull request merged**
